looker/resource_looker_user_attribute_group_test.go

```diff
@@ -0,0 +1,102 @@
+package looker
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	sdk "github.com/looker-open-source/sdk-codegen/go/sdk/v4"
+	"github.com/resolutionlife/terraform-provider-looker/internal/conv"
+)
+
+func init() {
+	// Add a sweeper to remove user attributes and groups that have names starting with `test_acc`.
+	resource.AddTestSweepers("looker_user_attribute_group", &resource.Sweeper{
+		Name: "looker_user_attribute_group",
+		F: func(r string) error {
+			c, err := newTestLookerSDK()
+			if err != nil {
+				return err
+			}
+
+			userAttrs, err := c.AllUserAttributes(sdk.RequestAllBoardSections{
+				Fields: conv.P(""),
+				Sorts:  conv.P(""),
+			}, nil)
+			if err != nil {
+				return err
+			}
+
+			filteredUserAttrs := make([]sdk.UserAttribute, 0, len(userAttrs))
+			for _, userAttr := range userAttrs {
+				if strings.HasPrefix(userAttr.Name, "test_acc_") {
+					filteredUserAttrs = append(filteredUserAttrs, userAttr)
+				}
+			}
+
+			for _, filteredUserAttr := range filteredUserAttrs {
+				if _, err := c.DeleteUserAttribute(*filteredUserAttr.Id, nil); err != nil {
+					return err
+				}
+			}
+
+			groups, err := c.SearchGroups(sdk.RequestSearchGroups{
+				Name: conv.PString("test-acc%"),
+			}, nil)
+			if err != nil {
+				return err
+			}
+
+			for _, g := range groups {
+				if _, err := c.DeleteGroup(*g.Id, nil); err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	})
+}
+
+func TestAccLookerUserAttributeGroup(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+				resource "looker_group" "test_acc" {
+					name = "test-acc-group"
+				}
+
+				resource "looker_user_attribute" "test_acc" {
+					name             = "test_acc_user_attribute_name"
+					label            = "test-acc-user-attribute-label"
+					data_type        = "number"
+					hidden           = true
+					default_value    = 24
+					user_access      = "View"
+					domain_whitelist = ["my_domain/route/sub/*"]
+				}
+
+				resource "looker_user_attribute_group" "test_acc" {
+					user_attribute_id = looker_user_attribute.test_acc.id
+					group_values {
+					  group_id = looker_group.test_acc.id
+					  value    = "25"
+					}
+				  }
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("looker_user_attribute_group.test_acc", "user_attribute_id"),
+					resource.TestCheckTypeSetElemNestedAttrs(
+						"looker_user_attribute_group.test_acc",
+						"group_values.*",
+						map[string]string{
+							"value": "25",
+						},
+					),
+				),
+			},
+		},
+	})
+}
```

Comment:
Just thinking about verifying it has been built, you could maybe use the `AllUserAttributeGroupValues` method? Something like this? :)
```suggestion

func testAccUserAttributeGroup(userAttrResource, groupResource string) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		userAttrRes, ok := s.RootModule().Resources[userAttrResource]
		if !ok {
			return errors.Errorf("Not found: %s", userAttrResource)
		}
		if userAttrRes.Primary.ID == "" {
			return errors.New("user attribiute ID is not set")
		}

		groupRes, ok := s.RootModule().Resources[groupResource]
		if !ok {
			return errors.Errorf("Not found: %s", groupRes)
		}
		if groupRes.Primary.ID == "" {
			return errors.New("group ID is not set")
		}

		client := testAccProvider.Meta().(*sdk.LookerSDK)

		userAttrs, err := client.AllUserAttributeGroupValues(userAttrRes.Primary.ID, "", nil)
		if err != nil {
			return errors.Wrapf(err, "failed to retrieve user attribute group value with id: %v", userAttrRes.Primary.ID)
		}

		for _, ua := range userAttrs {
			if *ua.GroupId == groupRes.Primary.ID {
				return nil
			}
		}

		return errors.Errorf("user attribute %s not found on group %s", userAttrRes.Primary.ID, groupRes.Primary.ID)
	}
}

```